File: C++/1007_MinimumDominoRotationsForEqualRow.cpp
class Solution {
public:
    int minDominoRotations(vector<int>& tops, vector<int>& bottoms) {
        int topTop = 1, bottomBottom = 1, topBottom = 0, bottomTop = 0, topSame = 0, bottomSame = 0, top = tops[0], bottom = bottoms[0];
        for(int i = 1; i < tops.size(); i++){
            if (tops[i] == top && bottoms[i] == top){
                topTop += 1;
                topBottom += 1;
                topSame += 1;
            } else if (tops[i] == top){
                topTop += 1;
            } else if (bottoms[i] == top){
                topBottom += 1;
            }
            if (tops[i] == bottom && bottoms[i] == bottom){
                bottomTop += 1;
                bottomBottom += 1;
                bottomSame += 1;
            } else if (tops[i] == bottom){
                bottomTop += 1;
            } else if (bottoms[i] == bottom){
                bottomBottom += 1;
            }
        }
        if (topTop + topBottom - topSame == bottoms.size() && bottomTop + bottomBottom - bottomSame == bottoms.size()){
            return min({topTop - topSame, topBottom-topSame, bottomTop - bottomSame, bottomBottom - bottomSame});
        }
        else if (topTop + topBottom - topSame == bottoms.size()){
            return min(topTop - topSame, topBottom-topSame);
        } else if (bottomTop + bottomBottom - bottomSame == bottoms.size()){
            return min(bottomTop - bottomSame, bottomBottom - bottomSame);
        } else{
            return -1;
        }
    }
};

File: 1007_MinimumDominoRotationsForEqualRow.py
class Solution:
    def minDominoRotations(self, tops: List[int], bottoms: List[int]) -> int:
        topTop = bottomBottom = 1
        topBottom = bottomTop = topSame = bottomSame = 0
        for i in range(1, len(tops)):
            if tops[i] == tops[0] and bottoms[i] == tops[0]:
                topTop += 1
                topBottom += 1
                topSame += 1
            elif tops[i] == tops[0]: 
                topTop += 1
            elif bottoms[i] == tops[0]:
                topBottom += 1
            if tops[i] == bottoms[0] and bottoms[i] == bottoms[0]:
                bottomBottom += 1
                bottomTop += 1
                bottomSame += 1
            elif bottoms[i] == bottoms[0]:
                bottomBottom += 1
            elif tops[i] == bottoms[0]:
                bottomTop += 1
        if (topTop + topBottom - topSame == len(tops)) and (bottomBottom + bottomTop - bottomSame == len(tops)):
            return min(topTop-topSame, topBottom-topSame, bottomBottom-bottomSame, bottomTop-bottomSame)
        elif topTop + topBottom - topSame == len(tops):
            return min(topTop- topSame, topBottom- topSame)
        elif bottomBottom + bottomTop - bottomSame == len(tops):
            return min(bottomBottom- bottomSame, bottomTop- bottomSame)
        else:
            return -1